doc/source/schema/schema.md

```diff
@@ -36,33 +39,73 @@ The terminology used in this specification is based on [xarray's terminology](ht
 * Coordinate values can be repeated and should not necessarily be numerical values.
 * Variables must not share names with dimensions.
-* Moreover, each group contains the following attributes:
-  - `created_at`: the date of creation of the group.
-  - `inference_library`: the library used to run the inference.
-  - `inference_library_version`: version of the inference library used.
+* Groups, variables or the InferenceData itself can have arbitrary metadata stored.
 
-### Relations
-`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations.
+### Metadata
+No metadata is _required_ to be present in order to be compliant with the InferenceData schema.
+However, it is recommended to store the following fields when relevant:
+* `name`: InferenceData objects represent multiple quantities related to Bayesian modelling,
+  but they are all tied to a single model. The model identifier can be added as metadata
+  to simplify the calls to model comparison functions.
+* `created_at`: the date of creation of the group.
+* `arviz_version`: the version of the ArviZ library that generated the InferenceData
+* `arviz_language`: the programming language from which ArviZ was used to create the InferenceData
+* `inference_library`: the library used to run the inference.
+* `inference_library_version`: version of the inference library used.
+
+
+### Relations between groups
+`InferenceData` data objects contain any combination of the groups described below. There are also some relations (detailed below) between the variables and dimensions of different groups. Hence, whenever related groups are present they should comply with these relations. Neither the presence of groups nor described below or the lack of some of the groups described below go against the schema.
 
 #### `posterior`
-Samples from the posterior distribution p(theta|y).
+Samples from the posterior distribution p(theta|y) in the parameter (also called constrained) space.
+
+(schema/unconstrained_posterior)=
+#### `unconstrained_posterior`
+Samples from the posterior distribution p(theta_transformed|y) in the unconstrained (also called transformed) space.
+
+Only variables that undergo a transformation for sampling should be present here.
+Therefore, to get the samples for _all_ the variables in the unconstrained space,
+variables should be taken from the `unconstrained_posterior` group if present,
+and if not, then the values from the variable in the `posterior` group should be used.
+
+Both samples and variables (for those present only) should match between the `posterior` and the `unconstrained_posterior` groups. Note that as defined above, matching samples and variables impose constraints on dimensions and coordinates, not on the values which will be different.
```

Comment:
What are these constraints?

E.g. constrained and unconstrained spaces might have different number of dimensions

Reply:
I was completely unaware of that 😓, so they would have the same sampling dimensions and variable name but not necessarly the same coords and dims, is that right? Would it be ok to allow for variables in unconstrained_posterior to have different shape than that same variable in posterior group, but that if they do have the same shape, then dims and coords should be shared?

Reply:
That sounds like it should work

e.g https://mc-stan.org/docs/reference-manual/simplex-transform.html

cc @avehtari 

Aki do you remember if any constrained vs unconstrained transform had more complicated structure?

Reply:
> Aki do you remember if any constrained vs unconstrained transform had more complicated structure?

Correlation matrices, Covariance matrices, Cholesky factors of covariance matrices, and Cholesky factors of correlation matrices https://mc-stan.org/docs/reference-manual/variable-transforms.html